### PR TITLE
[ISSUE-709] Improve release workflow

### LINKED ISSUE #709: Improve release workflow
**Is your feature request related to a problem? Please describe.**
Need to support release creation from release branch, not only from master.

**Describe the solution you'd like**
Chosen on the release creation page branch is used in a whole workflow in both repos.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by misanthropicat**
### Resolves #709

Supported choice of branch to release from.

## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [ ] New unit tests added
- [x] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
_Provide test details_

### CODE REVIEW

**Review by rkiselenko — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -10,33 +10,37 @@ jobs:
...
-
       - name: Parse release description
         id: split_description
         uses: jungwinter/split@v2
         with:
           msg: ${{ github.event.release.body }}
 
-      - name: Set versions
-        id: versions
+      - name: Set vars
```
Comment (safronovD) [inline .github/workflows/release.yml:19]: Should we update docs here?

Comment (misanthropicat) [reply .github/workflows/release.yml:19]: No, [docs ](https://github.com/dell/csi-baremetal-operator/blob/gh-pages/docs/index.yaml) are updated in operator repo: https://github.com/dell/csi-baremetal-operator/blob/b410ffb9cdaa48d8b1e680ff22d54036293c628d/.github/workflows/release.yml#L128

Comment (safronovD) [reply .github/workflows/release.yml:19]: I mean https://github.com/dell/csi-baremetal/blob/master/docs/release-workflow.md

Comment (safronovD) [reply .github/workflows/release.yml:19]: Seems like it is not related now and has no info about actions' usage

Comment (misanthropicat) [reply .github/workflows/release.yml:19]: Oh, I got it now, ok, I'll upgrade it, thanks.

Comment (misanthropicat) [reply .github/workflows/release.yml:19]: Doc is updated.

Comment (safronovD) [reply .github/workflows/release.yml:19]: thx!

**Pull request merged**